src/Compilers/VisualBasic/Portable/Symbols/Source/SourceDelegateClonedParameterSymbolForBeginAndEndInvoke.vb

```diff
@@ -0,0 +1,39 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+    Friend NotInheritable Class SourceDelegateClonedParameterSymbolForBeginAndEndInvoke
+        Inherits SourceClonedParameterSymbol
+
+        Public Sub New(originalParam As SourceParameterSymbol, newOwner As MethodSymbol, ordinal As Integer)
+            MyBase.New(originalParam, newOwner, ordinal)
+            Debug.Assert(Not originalParam.IsOptional)
+        End Sub
+
+        Friend Overrides ReadOnly Property IsCallerLineNumber As Boolean
+            Get
+                Return _originalParam.IsCallerLineNumber
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property IsCallerMemberName As Boolean
+            Get
+                Return _originalParam.IsCallerMemberName
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property IsCallerFilePath As Boolean
+            Get
+                Return _originalParam.IsCallerFilePath
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property CallerArgumentExpressionParameterIndex As Integer
+            Get
+                ' This parameter cannot be optional. Hence, this property shouldn't be accessed.
```

Comment:
>' This parameter cannot be optional. Hence, this property shouldn't be accessed. [](http://example.com/codeflow?start=16&length=80)

This assumption feels suspicious. An error doesn't prevent creation of symbols and doesn't prevent an attempt to bind method calls. We need to have tests proving the code path is unreachable.

Reply:
@AlekseyTs See `CheckDelegateParameterModifier`:

https://github.com/dotnet/roslyn/blob/5f06615edfb5c359c6dc1c75a5677eaddb0dba52/src/Compilers/VisualBasic/Portable/Binding/Binder_Delegates.vb#L923-L944

These are the flags used to create this symbol. If there was `Optional` flag, a diagnostic is produced **and** the Optional flag is removed. That's why I'm asserting non-optional parameter in the constructor itself.